pkg/kubelet/kubelet.go

```diff
@@ -2444,7 +2454,9 @@ func (kl *Kubelet) tryUpdateNodeStatus() error {
 		return fmt.Errorf("no node instance returned for %q", kl.nodeName)
 	}
 	kl.networkConfigMutex.Lock()
-	kl.podCIDR = node.Spec.PodCIDR
+	if kl.reconcileCIDR {
+		kl.podCIDR = node.Spec.PodCIDR
```

Comment:
The nodecontroller will write a cidr into the nodespec for the master kubelet as it does not differentiate it from other kubelets. We don't want master kubelet to use this cidr because this causes the crashloop described here:

https://github.com/kubernetes/kubernetes/issues/13102#issuecomment-134460647